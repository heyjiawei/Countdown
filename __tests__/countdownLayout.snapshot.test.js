import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import BlockLayout from '../components/CountdownBlockLayout';
import InlineBlockLayout from '../components/CountdownInlineBlockLayout';

describe('Countdown Block/InlineBlock Layout', () => {
  it('should render BlockLayout correctly', () => {
    const props = {id: '1', title: 'Driving Practice', days: 365, color: '#8bc34a', toShow: false};
    const clickHandler = jest.fn();
    const output = shallow(
      <BlockLayout
        title={props.title}
        days={props.days}
        color={props.color}
        onClick={clickHandler}
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render InlineBlockLayout correctly', () => {
    const props = {id: '1', title: 'Driving Practice', days: 36, color: '#8bc34a', toShow: false};
    const clickHandler = jest.fn();
    const output = shallow(
      <InlineBlockLayout
        title={props.title}
        days={props.days}
        color={props.color}
        onClick={clickHandler}
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render number of days, title and color correctly', () => {
    const props = {id: '1', title: 'Driving Practice', days: 365, color: '#8bc34a', toShow: false};
    const clickHandler = jest.fn();
    const blockLayout = shallow(
      <BlockLayout
        title={props.title}
        days={props.days}
        color={props.color}
        onClick={clickHandler}
      />
    );
    expect(blockLayout.find('span').at(0).text()).toEqual('365');
    expect(blockLayout.find('span').at(1).text()).toEqual('Driving Practice');
    expect(blockLayout.find('div').get(1).props.style.backgroundColor).toEqual('#8bc34a');
  });
  // can you test window resize? (Probably using mount from enzyme)
});
