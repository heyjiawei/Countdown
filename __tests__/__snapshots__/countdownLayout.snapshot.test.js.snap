// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Countdown Block/InlineBlock Layout should render BlockLayout correctly 1`] = `
<div
  className="toggle-option"
  onClick={[Function]}
>
  <div
    style={
      Object {
        "backgroundColor": "#8bc34a",
        "color": "white",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": "larger",
        "height": "36px",
        "lineHeight": "35px",
        "marginRight": "1%",
        "padding": "0em 0.5em",
        "width": "99%",
      }
    }
  >
    <span>
      365
    </span>
    <span
      style={
        Object {
          "fontSize": "smaller",
          "opacity": 0.8,
          "padding": "0em 1em",
        }
      }
    >
      Driving Practice
    </span>
  </div>
</div>
`;

exports[`Countdown Block/InlineBlock Layout should render InlineBlockLayout correctly 1`] = `
<div
  className="toggle-option"
  onClick={[Function]}
  style={
    Object {
      "fontFamily": "Roboto, sans-serif",
      "fontSize": "larger",
      "height": "36px",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "#8bc34a",
        "color": "white",
        "display": "inline-block",
        "height": "36px",
        "lineHeight": "35px",
        "marginRight": "1%",
        "padding": "0em 0.5em",
        "width": "9.863013698630137%",
      }
    }
  >
    <span>
      36
    </span>
  </div>
  <div
    style={
      Object {
        "display": "inline",
        "fontSize": "smaller",
        "opacity": 0.6,
      }
    }
  >
    <span>
      Driving Practice
    </span>
  </div>
</div>
`;
